lua/bufferline/render.lua

```diff
@@ -520,17 +519,22 @@ function render.enable()
       end
 
       local bufnames = {}
-      for _,bufnr in ipairs(state.buffers) do
+      for _, bufnr in ipairs(state.buffers) do
         local name = buf_get_name(bufnr)
         if use_relative_file_paths then
           name = utils.relative(name)
         end
+
         -- escape quotes
         name = name:gsub('"', '\\"')
-        bufnames[#bufnames + 1] = '"' .. name .. '"'
+
+        bufnames[#bufnames + 1] = '{' ..
```

Comment:
Implementation part 1. Instead of passing `{"name", "name2", "name3"}` to `restore_buffers`, we pass:

```lua
{
  {name = "a", pinned = false},
  {name = "b", pinned = true},
  {name = "c", pinned = false},
}
```

This can be extended to allow for more fields in the future.